### PR TITLE
Switch from `cgmath` to `glam` in examples

### PR DESCRIPTION
**Pull request by a1phyr**
Switch from `cgmath` to `glam` in examples.

There are now a few degrees to radians conversions, it might be worth switching everything to radians.

**Testing**
Ran changed examples

### CODE REVIEW

File: wgpu/examples/framework.rs [hunk 1]
```diff
@@ -10,15 +10,6 @@ use winit::{
 #[path = "../tests/common/mod.rs"]
 pub mod test_common;
 
-#[rustfmt::skip]
-#[allow(unused)]
-pub const OPENGL_TO_WGPU_MATRIX: cgmath::Matrix4<f32> = cgmath::Matrix4::new(
```
Comment (kvark) [inline wgpu/examples/framework.rs:15]: hoorrray!

File: wgpu/examples/mipmap/main.rs [hunk 2]
```diff
@@ -61,15 +61,14 @@ struct Example {
...
 
 impl Example {
-    fn generate_matrix(aspect_ratio: f32) -> cgmath::Matrix4<f32> {
-        let mx_projection = cgmath::perspective(cgmath::Deg(45f32), aspect_ratio, 1.0, 1000.0);
-        let mx_view = cgmath::Matrix4::look_at_rh(
-            cgmath::Point3::new(0f32, 0.0, 10.0),
-            cgmath::Point3::new(0f32, 50.0, 0.0),
-            cgmath::Vector3::unit_z(),
+    fn generate_matrix(aspect_ratio: f32) -> glam::Mat4 {
+        let projection = glam::Mat4::perspective_rh(consts::FRAC_PI_4, aspect_ratio, 1.0, 1000.0);
```
Comment (kvark) [inline wgpu/examples/mipmap/main.rs:65]: do you remember if the projected depth of this glam function is for 0 to 1, or for -1 to 1?

**Review by kvark — COMMENTED** [review]
Nicely done!

File: wgpu/examples/mipmap/main.rs (hunk 2 above)
Comment (a1phyr) [reply wgpu/examples/mipmap/main.rs:65]: [According to the doc](https://docs.rs/glam/latest/glam/f32/struct.Mat4.html#method.perspective_rh), it's from 0 to 1. There is `perspective_rh_gl` for -1 to 1.

File: wgpu/examples/cube/main.rs [hunk 3]
```diff
@@ -111,15 +111,14 @@ struct Example {
...
-        let mx_projection = cgmath::perspective(cgmath::Deg(45f32), aspect_ratio, 1.0, 10.0);
-        let mx_view = cgmath::Matrix4::look_at_rh(
-            cgmath::Point3::new(1.5f32, -5.0, 3.0),
-            cgmath::Point3::new(0f32, 0.0, 0.0),
-            cgmath::Vector3::unit_z(),
+    fn generate_matrix(aspect_ratio: f32) -> glam::Mat4 {
+        let projection = glam::Mat4::perspective_rh(consts::FRAC_PI_4, aspect_ratio, 1.0, 10.0);
+        let view = glam::Mat4::look_at_rh(
+            glam::Vec3::new(1.5f32, -5.0, 3.0),
+            glam::Vec3::new(0f32, 0.0, 0.0),
```
Comment (cwfitzgerald) [inline wgpu/examples/cube/main.rs:118]: `Vec3::ZERO` can be used here

File: wgpu/examples/water/main.rs [hunk 4]
```diff
@@ -99,31 +95,29 @@ impl Example {
...
 
         let reg_view = reg_view * scale;
 
-        let flipped_view = cgmath::Matrix4::look_at_rh(
-            cgmath::Point3::new(CAMERA.x, -CAMERA.y, CAMERA.z),
-            cgmath::Point3::new(0f32, 0.0, 0.0),
-            cgmath::Vector3::unit_y(),
+        let flipped_view = glam::Mat4::look_at_rh(
+            glam::Vec3::new(CAMERA.x, -CAMERA.y, CAMERA.z),
+            glam::Vec3::new(0f32, 0.0, 0.0),
```
Comment (cwfitzgerald) [inline wgpu/examples/water/main.rs:111]:
```suggestion
            glam::Vec3::ZERO,
```

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Small nits, but otherwise LGTM

**Comment by a1phyr** [comment]
What should we do with the failing `skybox` test ?

It used to multiply a view matrix by `OPENGL_TO_WGPU_MATRIX`, which is weird, but now I removed it, it fails.

**Comment by cwfitzgerald** [comment]
If the output looks visually right, you need to remove the screenshots in the example directory then run `cargo test --example skybox` to regenerate them.

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**